### ISSUE TITLE
`prefer-some-in-iteration` false positive in set entries list

### ISSUE THREAD

**Issue by anderseknert** [StyraInc/regal#553]
Doing something like the below should likely be a linter issue in its own right :) But Regal should definitely not recommend using `some ... in` style iteration here:

```rego
package p

my_set := {
    false,
    input.arr[_] == "bar"
}
```

(Obviously not limited to set items, but any similar context)

**Comment by srenatus** [comment]
`"bar" in input.arr` should work in that place, though. "Some" is for introducing new bindings.

**Comment by anderseknert** [comment]
Yep, hence the false positive. I guess a next step would be to have the use-in-operator rule catch the old style iteration here, but false positives are worse than something not reported.
